app/repositories/QuestionnaireRepository.scala

```diff
@@ -35,7 +35,8 @@ import scala.language.postfixOps
 trait QuestionnaireRepository {
   def addQuestions(applicationId: String, questions: List[PersistedQuestion]): Future[Unit]
   def findQuestions(applicationId: String): Future[Map[String, String]]
-  def onlineTestPassMarkReport: Future[Map[String, PassMarkReportQuestionnaireData]]
+  def findForOnlineTestPassMarkReport: Future[Map[String, QuestionnaireReportItem]]
+  def findAllForDiversityReport: Future[Map[String, QuestionnaireReportItem]]
```

Comment:
The problem I was trying to solve is that both findForOnlineTestPassMarkReport and findAllForDiversityReport return the same kind of objects but in one of them we do not filter, we return everything and in the other one we filter the results. I thought about having only one method because in findForOnlineTestPassMarkReport we probably don't need the following lines:

val firstEmploymentQuestion = "When you were 14, what kind of work did your highest-earning parent or guardian do?"
      val query = BSONDocument(s"questions.$firstEmploymentQuestion" -> BSONDocument("$exists" -> BSONBoolean(true)))

So then, both methods are equal.

I think that filter is just an optimisation, in that case we can call it findAll, although even in that case I would prefer findAllAsReportItem and I would reserve findAll to a method that returns the persisted questionnaire object.